Setting 'text.usetex : True' in matplotlibrc causes some tests to fail

If I set 'test.usetex : True' in my matplotlibrc, some astropy tests for the visualization package fail, one example output (full test dump linked below):

```
>       assert ax.coords[1].format_coord(4) == '4\xb000\'00\"'
E       assert '$4^\\circ00{...rime\\prime}$' == '4°00\'00"'
E         - $4^\circ00{}^\prime00{}^{\prime\prime}$
E         + 4°00'00"
```

[errors.log](https://github.com/astropy/astropy/files/2270271/errors.log)

Running in a conda virtual environment:

```
Python 3.6.6 |Anaconda, Inc.| (default, Jun 28 2018, 17:14:51) 
[GCC 7.2.0] on linux

# packages in environment at /Users/users/koman/.conda/envs/astropy-dev:
#
# Name                    Version                   Build  Channel
astropy                   3.1.dev22306              <pip>
atomicwrites              1.1.5                    py36_0  
attrs                     18.1.0                   py36_0  
blas                      1.0                         mkl  
ca-certificates           2018.03.07                    0  
certifi                   2018.4.16                py36_0  
cycler                    0.10.0                   py36_0  
cython                    0.28.4           py36hf484d3e_0  
dbus                      1.13.2               h714fa37_1  
expat                     2.2.5                he0dffb1_0  
fontconfig                2.13.0               h9420a91_0  
freetype                  2.9.1                h8a8886c_0  
glib                      2.56.1               h000015b_0  
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb453b48_1  
h5py                      2.8.0            py36h8d01980_0  
hdf5                      1.10.2               hba1933b_1  
icu                       58.2                 h9c2bf20_1  
intel-openmp              2018.0.3                      0  
jpeg                      9b                   h024ee3a_2  
kiwisolver                1.0.1            py36hf484d3e_0  
libedit                   3.1.20170329         h6b74fdf_2  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 7.2.0                hdf63c60_3  
libgfortran-ng            7.2.0                hdf63c60_3  
libpng                    1.6.34               hb9fc6fc_0  
libstdcxx-ng              7.2.0                hdf63c60_3  
libuuid                   1.0.3                h1bed415_2  
libxcb                    1.13                 h1bed415_1  
libxml2                   2.9.8                h26e45fe_1  
matplotlib                2.2.2            py36hb69df0a_2  
mkl                       2018.0.3                      1  
mkl_fft                   1.0.4            py36h4414c95_1  
mkl_random                1.0.1            py36h4414c95_1  
more-itertools            4.3.0                    py36_0  
ncurses                   6.1                  hf484d3e_0  
numpy                     1.15.0           py36h1b885b7_0  
numpy-base                1.15.0           py36h3dfced4_0  
openssl                   1.0.2o               h14c3975_1  
pandas                    0.23.4           py36h04863e7_0  
pcre                      8.42                 h439df22_0  
pip                       10.0.1                   py36_0  
pip                       18.0                      <pip>
pluggy                    0.6.0                    py36_0  
psutil                    5.4.6            py36h14c3975_0  
py                        1.5.4                    py36_0  
pyparsing                 2.2.0                    py36_1  
pyqt                      5.9.2            py36h22d08a2_0  
pytest                    3.6.4                    py36_0  
pytest-arraydiff          0.2              py36h39e3cac_0  
pytest-astropy            0.4.0                    py36_0  
pytest-doctestplus        0.1.3                    py36_0  
pytest-openfiles          0.3.0                    py36_0  
pytest-remotedata         0.3.0                    py36_0  
python                    3.6.6                hc3d631a_0  
python-dateutil           2.7.3                    py36_0  
pytz                      2018.5                   py36_0  
pyyaml                    3.13             py36h14c3975_0  
qt                        5.9.6                h52aff34_0  
readline                  7.0                  ha6073c6_4  
scipy                     1.1.0            py36hc49cb51_0  
setuptools                39.2.0                   py36_0  
sip                       4.19.8           py36hf484d3e_0  
six                       1.11.0                   py36_1  
sqlite                    3.24.0               h84994c4_0  
tk                        8.6.7                hc745277_3  
tornado                   5.0.2            py36h14c3975_0  
wheel                     0.31.1                   py36_0  
xz                        5.2.4                h14c3975_4  
yaml                      0.1.7                had09818_2  
zlib                      1.2.11               ha838bed_2  

```

--- a/astropy/conftest.py
+++ b/astropy/conftest.py
@@ -45,8 +45,7 @@ def pytest_configure(config):
 def pytest_unconfigure(config):
     # do not assign to matplotlibrc_cache in function scope
     if HAS_MATPLOTLIB:
-        for k, v in matplotlibrc_cache.items():
-            matplotlib.rcParams[k] = v
+        matplotlib.rcParams.update(matplotlibrc_cache)
         matplotlibrc_cache.clear()
 
 

--- a/astropy/conftest.py
+++ b/astropy/conftest.py
@@ -9,6 +9,12 @@ from importlib.util import find_spec
 from astropy.tests.plugins.display import PYTEST_HEADER_MODULES
 from astropy.tests.helper import enable_deprecations_as_exceptions
 
+try:
+    import matplotlib
+except ImportError:
+    HAS_MATPLOTLIB = False
+else:
+    HAS_MATPLOTLIB = True
 
 if find_spec('asdf') is not None:
     from asdf import __version__ as asdf_version
@@ -23,34 +29,25 @@ enable_deprecations_as_exceptions(
     # installed. This can be removed once pyopenssl 1.7.20+ is released.
     modules_to_ignore_on_import=['requests'])
 
-try:
-    import matplotlib
-except ImportError:
-    pass
-else:
+if HAS_MATPLOTLIB:
     matplotlib.use('Agg')
-
 
 matplotlibrc_cache = {}
 
+
 def pytest_configure(config):
-    try:
-        import matplotlib
-    except ImportError:
-        pass
-    else:
+    # do not assign to matplotlibrc_cache in function scope
+    if HAS_MATPLOTLIB:
         matplotlibrc_cache.update(matplotlib.rcParams)
         matplotlib.rcdefaults()
 
+
 def pytest_unconfigure(config):
-    try:
-        import matplotlib
-    except ImportError:
-        pass
-    else:
+    # do not assign to matplotlibrc_cache in function scope
+    if HAS_MATPLOTLIB:
         for k, v in matplotlibrc_cache.items():
             matplotlib.rcParams[k] = v
         matplotlibrc_cache.clear()
 
-    
+
 PYTEST_HEADER_MODULES['Cython'] = 'cython'

--- a/astropy/conftest.py
+++ b/astropy/conftest.py
@@ -30,4 +30,27 @@ except ImportError:
 else:
     matplotlib.use('Agg')
 
+
+matplotlibrc_cache = {}
+
+def pytest_configure(config):
+    try:
+        import matplotlib
+    except ImportError:
+        pass
+    else:
+        matplotlibrc_cache.update(matplotlib.rcParams)
+        matplotlib.rcdefaults()
+
+def pytest_unconfigure(config):
+    try:
+        import matplotlib
+    except ImportError:
+        pass
+    else:
+        for k, v in matplotlibrc_cache.items():
+            matplotlib.rcParams[k] = v
+        matplotlibrc_cache.clear()
+
+    
 PYTEST_HEADER_MODULES['Cython'] = 'cython'

